core/shared/src/main/scala/hammock/Uri.scala

```diff
@@ -30,17 +31,50 @@ import cats.syntax.show._
   * @param fragment  fragment of the uri. For example #header1
   */
 case class Uri(
-    scheme: Option[Scheme] = None,
-    authority: Option[Authority] = None,
-    path: String = "",
-    query: Map[String, String] = Map(),
-  fragment: Option[Fragment] = None) {
+                scheme: Option[Scheme] = None,
+                authority: Option[Authority] = None,
+                path: String = "",
+                query: Map[String, String] = Map(),
+                fragment: Option[Fragment] = None) {
 
   /** Append a string to the path of the [[Uri]]
     */
   def /(str: String): Uri = {
     copy(path = s"$path/$str")
   }
+
+  /**
+    * Append query parameter to [[query]]
+    *
+    * @param key   - parameter name
+    * @param value - the value
+    * @return updated [[Uri]]
+    **/
+  def param(key: String, value: String): Uri = copy(query = this.query + (key → value))
+
+  /**
+    * Appends multiple query parameters to [[query]]
+    *
+    * @param ps - parameters
+    * @return updated [[Uri]]
+    **/
+  def params(ps: (String, String)*): Uri = ps match {
+    case Seq() ⇒ this
+    case _     ⇒ ps.foldLeft(this) { case (uri, (k, v)) ⇒ uri.copy(query = uri.query + (k → v)) }
+  }
+
+  /**
+    * Produces the same result as [[params]]
+    * but provides syntax as you are writing URI query in browser
+    * Usage example:
+    * {{{
+    *   uri"example.com" ? (("a" → "b") & ("c" → "d") & ("e" → "f"))
+    * }}}
+    *
+    * @param ps - parameters
+    * @return updated [[Uri]]
+    **/
+  def ?(ps: Seq[(String, String)]): Uri = params(ps: _*)
```

Comment:
Since `Seq` can be mutable, can you specify it to a more suitable data structure, such as `List` or `Vector`?